vowpalwabbit/vwdll.cpp

```diff
@@ -149,7 +149,7 @@ VW_DLL_PUBLIC void VW_CALLING_CONV VW_EndParser(VW_HANDLE handle)
 VW_DLL_PUBLIC VW_EXAMPLE VW_CALLING_CONV VW_GetExample(VW_HANDLE handle)
 { vw * pointer = static_cast<vw*>(handle);
   parser * parser_pointer = static_cast<parser *>(pointer->example_parser);
-  return static_cast<VW_EXAMPLE>(VW::get_example(parser_pointer));
+  return static_cast<VW_EXAMPLE>(VW::get_example(*pointer));
```

Comment:
don't touch anything in this file. This is our DLL binding layer and any changes here can be dangerous

Reply:
If I were to change the parameter types of the function `get_example`, then without making the corresponding change here, the compiler gives an error. So, I should avoid changing the parameter types of functions used here?

Reply:
In general, changing any code in a dynamic library's binding layer can be extremely problematic. There are things you can safely change, but it generally requires extreme care. Since this is in a .cpp file, its _probably_ safe, but I currently don't have the time to go in and analyze the implications of any changes made here. So in general, its best to simply avoid changing any of the code explicitly exposed in the APIs.

If you feel its absolutely necessary, feel free to create new functions (add their prototypes and definitions in their own file if you need to). We can go back and fix it up later once a full evaluation's been done